### ISSUE TITLE
EmrServerlessStartJobOperator does not raise airflow exception

### ISSUE THREAD

**Issue by EzuA** [aws-samples/emr-serverless-samples#22]
Hi! I am testing the Operator and I found that airflow is marking a task as `SUCCESS` even though the EMR job state is `FAILED`.
I think the problem is in the `EmrServerlessHook`, this condition:

```           
if state in failure_states:
    raise AirflowException(
        f"{object_type.title()} reached failure state {state}."
    )
```

should be at the end in the while loop. 

Thanks!

**Comment by dacort** [comment]
Thanks @EzuA - I'll look into this today!

**Comment by fventuri-availity** [comment]
I think it is because this line (https://github.com/aws-samples/emr-serverless-samples/blob/main/airflow/emr_serverless/operators/emr.py#L220) has the 'desired_state' set to 'EmrServerlessJobSensor.TERMINAL_STATES', while I think it should have 'EmrServerlessJobSensor.SUCCESS_STATES':
```
                desired_state=EmrServerlessJobSensor.SUCCESS_STATES,
```

Franco

**Comment by dacort** [comment]
Yep, I think you're right, thank you!. :) Was just tracing the logic to verify.

**Comment by EzuA** [comment]
Yes, I have tested both options and they are working in my local environment. I think the option proposed by Franco is the best.

**Comment by dacort** [comment]
Testing this locally now as well - should have an updated release by end of day. Thanks to both of you!

**Comment by dacort** [comment]
Thanks again - fixed in [v1.0.1](https://github.com/aws-samples/emr-serverless-samples/releases/tag/v1.0.1).
